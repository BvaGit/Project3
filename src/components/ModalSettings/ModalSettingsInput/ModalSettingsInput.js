import React from 'react';

import Input from '../../Input';

const ModalSettingsInput = ({ id, name, labelId, type, placeholderId, changeInput }) => {

    const handleChange = (e) => {
        changeInput({ [id]:e.target.value })  
    }

    return(
        <Input 
            id={id} 
            name={name}
            labelId={labelId} 
            type={type} 
            placeholderId={placeholderId} 
            onChange={handleChange}
        />
    )
}

export default ModalSettingsInput;