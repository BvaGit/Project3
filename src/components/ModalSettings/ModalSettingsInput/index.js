import ModalSettingsInput from "./ModalSettingsInput";
import { connect } from 'react-redux';
import { getSettingsLoginStore, getSettingsPasswordStore } from '../../../store/settings/selectors';
import { handleValueInput } from '../../../store/settings/actions';

const mapStateToProps = (state, props) => ({
    login: getSettingsLoginStore(state, props),
    password: getSettingsPasswordStore(state, props)
})
const mapDispatchToProps = (dispatch) => ({
    changeInput: (data) => dispatch(handleValueInput(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(ModalSettingsInput);
