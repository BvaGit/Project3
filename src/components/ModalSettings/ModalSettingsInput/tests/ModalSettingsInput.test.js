import React from 'react';
import { shallow, mount } from 'enzyme';
import ModalSettingsInput from '../ModalSettingsInput';
import { Provider } from 'react-redux';
import { IntlProvider } from "react-intl";


describe('Rooms', () => {   
    it("Should match snapshot", () => {
        
        const component = shallowBiggerSmart(<ModalSettingsInput />)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider >{core}</IntlProvider>);
};