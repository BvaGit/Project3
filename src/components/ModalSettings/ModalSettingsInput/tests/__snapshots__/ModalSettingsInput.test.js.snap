// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms Should match snapshot 1`] = `"<label class=\\"inputs__label \\">undefined<input title=\\"undefined\\" placeholder=\\"undefined\\" class=\\"inputs__input undefined\\"/></label>"`;
